app/controllers/settings/profiles_controller.rb

```diff
@@ -16,11 +16,35 @@ def update
       render :edit, status: :unprocessable_entity
     end
   end
+  
+  def destroy
+    begin
+      if Current.user.admin?
```

Comment:
This is much simpler I hope.
We just refuse to delete last admin with members - otherwise all goes through